### PR TITLE
Console/preset filtering with comments

### PR DESCRIPTION
**Pull request by delislej**
added second drop down to make the list less cluttered. filters by console selection with added console value in json files.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,3 +1,5 @@
 source 'https://rubygems.org'
 gem 'github-pages', '>=39'
 gem 'jekyll-sitemap'
+
+gem "webrick", "~> 1.7"
```
Comment (darthcloud) [inline Gemfile]: What is webrick used for?

File: blueretro_presets.js [hunk 2]
```diff
@@ -254,6 +277,7 @@ function fetchMap(presets, files, idx) {
             reject(error);
         });
     });
+
```
Comment (darthcloud) [inline blueretro_presets.js]: RM extra empty line

File: blueretro_presets.js [hunk 3]
```diff
@@ -284,6 +313,7 @@ function initBlueRetroCfg() {
     .catch(error => {
         log('Argh! ' + error);
     });
+
```
Comment (darthcloud) [inline blueretro_presets.js]: Same

File: blueretro_presets.js [hunk 4]
```diff
@@ -341,41 +371,46 @@ function writeInputCfg(cfgId, cfg) {
...
-    document.getElementById("inputSaveText").style.display = 'none';
+    //get consoleName and preset name
     var preset = Number(document.getElementById("presetsName").value);
-    var nbMapping = presets[preset].map.length;
-    var cfgSize = nbMapping*8 + 3;
-    var cfg = new Uint8Array(cfgSize);
-    var cfgId = Number(document.getElementById("inputSelect").value);
+    var consoleName = Number(document.getElementById("consoleName").value);
+    //make sure both console and preset are not placeholders before we do anything
+    if(preset != -1 && consoleName != -1){
```
Comment (darthcloud) [inline blueretro_presets.js]: Didn't had time to run this locally to see, but are the selection box empty now when page load?

File: blueretro_presets.js [hunk 5]
```diff
@@ -414,5 +449,41 @@ function btConn() {
 }
 
 function selectInput() {
-    document.getElementById("desc").textContent = presets[Number(this.value)].desc;
+    //only change the description if selected input is not the placeholder
+    if(Number(document.getElementById("presetsName").value) == -1 || Number(document.getElementById("consoleName").value) == -1)
```
Comment (darthcloud) [inline blueretro_presets.js]: Sorry to be picky about style, but make sure you got a space between the if (or any other statement) and the parentheses/braces. Same for curly bracket that open following the closing parenthesis.

File: map/jag_cycle_shift.json [hunk 6]
```diff
@@ -1,6 +1,7 @@
 {
     "name":"Jaguar cycle shift",
     "desc":"Workaround for Jaguar games with broken mapping.",
+    "console":"Jag",
```
Comment (darthcloud) [inline map/jag_cycle_shift.json]: Name should match https://github.com/darthcloud/BlueRetroWebCfg/blob/c7d8bea9fc5d80bc78268e69dd6e0a01e7de705d/blueretro.js#L168

I think this is only non-matching one, use Jaguar here

**Review by darthcloud — CHANGES_REQUESTED** [review]
Overall this look good, if you make change before weekend I could test it and merge it after at that time.

File: Gemfile (hunk 1 above)
Comment (delislej) [reply Gemfile]: webrick was a required dependency for me to run this locally, it can be removed. good catch.

File: blueretro_presets.js (hunk 2 above)
Comment (delislej) [reply blueretro_presets.js]: done

File: blueretro_presets.js (hunk 3 above)
Comment (delislej) [reply blueretro_presets.js]: done

File: map/jag_cycle_shift.json (hunk 6 above)
Comment (delislej) [reply map/jag_cycle_shift.json]: done

File: blueretro_presets.js (hunk 5 above)
Comment (delislej) [reply blueretro_presets.js]: double checked and changed all i could find to match the style. if you catch any more let me know!

File: blueretro_presets.js (hunk 4 above)
Comment (delislej) [reply blueretro_presets.js]: first box is output defaulting to Output 1, second box says select a console, third box was left blank until a console was selected. on second thought i feel having the third box say "select a preset" as place holder until a console is selected looks better. no saving can be done until both a console and a preset is selected.

image

changed to

image

the selection box text can be changed trivially. let me know what you prefer.

**Comment by delislej** [comment]
ok, all of the requested changes have been made! i almost forgot to change the getmaplist call back to your github. i needed to use mine to add and test the console filtering.

File: blueretro_presets.js (hunk 4 above)
Comment (darthcloud) [reply blueretro_presets.js]: I think 2nd one is better, also maybe let people skip console filter and make all map available in the preset list initially.

**Comment by delislej** [comment]
ok, on load  it looks like this:

image

the dropdown looks like this: 

image

when a console is selected it only shows that consoles presets:

image

when a preset is selected it updates the description:

image

when all is selected it resets the description and fill the preset menu with all the options again:

image

**Review by darthcloud — APPROVED** [review]

**Comment by darthcloud** [comment]
Merged thanks!

**Pull request merged**